### ISSUE TITLE
Regression starting tabula on a different port in 1.1.0-beta.1a

### ISSUE THREAD

**Issue by pdemilly** [tabulapdf/tabula#486]
When starting tabula on a different port on Linux I am getting this stacktrace. (This used to work with tabula 1.0

java -Dfile.encoding=utf-8 -Xms256M -Xmx1024M -Dwarbler.port=9999 -jar tabula.jar 

```
2016-03-31 17:19:26.884:INFO::main: Logging initialized @182ms
2016-03-31 17:19:26.890:INFO:oejr.Runner:main: Runner
2016-03-31 17:19:26.996:INFO:oejs.Server:main: jetty-9.2.9.v20150224
2016-03-31 17:19:32.374:WARN:oeja.AnnotationConfiguration:main: ServletContainerInitializers: detected. Class hierarchy: empty
INFO: jruby 1.7.24 (1.9.3p551) 2016-01-20 bd68d85 on OpenJDK 64-Bit Server VM 1.7.0_95-b00 +jit [linux-amd64]
INFO: using a shared (threadsafe!) runtime
DATA_DIR = /home/pascal/.tabula
DOCUMENTS_BASEPATH = /home/pascal/.tabula/pdfs
ENABLE_DEBUG_METHODS = false
============================================================
http://127.0.0.1:9999
============================================================

======================================================
Launching web browser to http://127.0.0.1:9999

If it does not open in 10 seconds, you may manually open
a web browser to the above URL.
When you're done using the Tabula interface, you may
return to this window and press "Control-C" to close it.
======================================================

ERROR: initialization failed
org.jruby.rack.RackInitializationException: java.io.IOException: Failed to show URI:http://127.0.0.1:9999
	at org.jruby.rack.RackInitializationException.wrap(RackInitializationException.java:31)
	at org.jruby.rack.RackApplicationFactoryDecorator.init(RackApplicationFactoryDecorator.java:104)
	at org.jruby.rack.RackServletContextListener.contextInitialized(RackServletContextListener.java:50)
	at org.eclipse.jetty.server.handler.ContextHandler.callContextInitialized(ContextHandler.java:800)
	at org.eclipse.jetty.servlet.ServletContextHandler.callContextInitialized(ServletContextHandler.java:444)
	at org.eclipse.jetty.server.handler.ContextHandler.startContext(ContextHandler.java:791)
	at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:294)
	at org.eclipse.jetty.webapp.WebAppContext.startWebapp(WebAppContext.java:1349)
	at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1342)
	at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:741)
	at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:505)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:132)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:114)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:61)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.doStart(ContextHandlerCollection.java:163)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:132)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:114)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:61)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:132)
	at org.eclipse.jetty.server.Server.start(Server.java:387)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:114)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:61)
	at org.eclipse.jetty.server.Server.doStart(Server.java:354)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at org.eclipse.jetty.runner.Runner.run(Runner.java:509)
	at org.eclipse.jetty.runner.Runner.main(Runner.java:557)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at WarMain.launchWebServer(WarMain.java:181)
	at WarMain.start(WarMain.java:355)
	at JarMain.doStart(JarMain.java:228)
	at WarMain.main(WarMain.java:378)
Caused by: java.io.IOException: Failed to show URI:http://127.0.0.1:9999
	at sun.awt.X11.XDesktopPeer.launch(sun/awt/X11/XDesktopPeer.java:114)
	at sun.awt.X11.XDesktopPeer.browse(sun/awt/X11/XDesktopPeer.java:98)
	at java.awt.Desktop.browse(java/awt/Desktop.java:386)
	at java.lang.reflect.Method.invoke(java/lang/reflect/Method.java:606)
	at RUBY.(root)(/tmp/jetty-0.0.0.0-9999-tabula.jar-_-any-747100388240792074.dir/webapp/WEB-INF/config.ru:51)
	at org.jruby.RubyBasicObject.instance_eval(org/jruby/RubyBasicObject.java:1594)
	at RUBY.initialize(/tmp/jetty-0.0.0.0-9999-tabula.jar-_-any-747100388240792074.dir/webapp/WEB-INF/gems/gems/rack-1.6.4/lib/rack/builder.rb:55)
	at RUBY.(root)(/tmp/jetty-0.0.0.0-9999-tabula.jar-_-any-747100388240792074.dir/webapp/WEB-INF/config.ru:1)
	at org.eclipse.jetty.server.handler.ContextHandler.callContextInitialized(org/eclipse/jetty/server/handler/ContextHandler.java:800)
	at org.eclipse.jetty.servlet.ServletContextHandler.callContextInitialized(org/eclipse/jetty/servlet/ServletContextHandler.java:444)
	at org.eclipse.jetty.server.handler.ContextHandler.startContext(org/eclipse/jetty/server/handler/ContextHandler.java:791)
	at org.eclipse.jetty.servlet.ServletContextHandler.startContext(org/eclipse/jetty/servlet/ServletContextHandler.java:294)
	at org.eclipse.jetty.webapp.WebAppContext.startWebapp(org/eclipse/jetty/webapp/WebAppContext.java:1349)
	at org.eclipse.jetty.webapp.WebAppContext.startContext(org/eclipse/jetty/webapp/WebAppContext.java:1342)
	at org.eclipse.jetty.server.handler.ContextHandler.doStart(org/eclipse/jetty/server/handler/ContextHandler.java:741)
	at org.eclipse.jetty.webapp.WebAppContext.doStart(org/eclipse/jetty/webapp/WebAppContext.java:505)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(org/eclipse/jetty/util/component/AbstractLifeCycle.java:68)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(org/eclipse/jetty/util/component/ContainerLifeCycle.java:132)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(org/eclipse/jetty/util/component/ContainerLifeCycle.java:114)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(org/eclipse/jetty/server/handler/AbstractHandler.java:61)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.doStart(org/eclipse/jetty/server/handler/ContextHandlerCollection.java:163)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(org/eclipse/jetty/util/component/AbstractLifeCycle.java:68)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(org/eclipse/jetty/util/component/ContainerLifeCycle.java:132)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(org/eclipse/jetty/util/component/ContainerLifeCycle.java:114)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(org/eclipse/jetty/server/handler/AbstractHandler.java:61)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(org/eclipse/jetty/util/component/AbstractLifeCycle.java:68)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(org/eclipse/jetty/util/component/ContainerLifeCycle.java:132)
	at org.eclipse.jetty.server.Server.start(org/eclipse/jetty/server/Server.java:387)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(org/eclipse/jetty/util/component/ContainerLifeCycle.java:114)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(org/eclipse/jetty/server/handler/AbstractHandler.java:61)
	at org.eclipse.jetty.server.Server.doStart(org/eclipse/jetty/server/Server.java:354)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(org/eclipse/jetty/util/component/AbstractLifeCycle.java:68)
	at org.eclipse.jetty.runner.Runner.run(org/eclipse/jetty/runner/Runner.java:509)
	at org.eclipse.jetty.runner.Runner.main(org/eclipse/jetty/runner/Runner.java:557)
	at java.lang.reflect.Method.invoke(java/lang/reflect/Method.java:606)
	... 4 more
2016-03-31 17:19:38.025:INFO:oejsh.ContextHandler:main: Started o.e.j.w.WebAppContext@46a329dc{/,file:/tmp/jetty-0.0.0.0-9999-tabula.jar-_-any-747100388240792074.dir/webapp/,AVAILABLE}{file:/home/pascal/tmp/tabula/tabula.jar}
2016-03-31 17:19:38.025:WARN:oejsh.RequestLogHandler:main: !RequestLog
2016-03-31 17:19:38.035:INFO:oejs.ServerConnector:main: Started ServerConnector@6cfb365c{HTTP/1.1}{0.0.0.0:9999}
2016-03-31 17:19:38.036:INFO:oejs.Server:main: Started @11336ms
```

**Comment by jeremybmerrill** [comment]
Odd... it's working for me. What Java version? What distro? Did it happen more than once?

`$ java -version
openjdk version "1.8.0_66-internal"
OpenJDK Runtime Environment (build 1.8.0_66-internal-b17)
OpenJDK 64-Bit Server VM (build 25.66-b17, mixed mode)`

and I'm on Ubuntu 15.10

**Comment by pdemilly** [comment]
java version "1.7.0_95"
OpenJDK Runtime Environment (IcedTea 2.6.4) (7u95-2.6.4-0ubuntu0.12.04.1)
OpenJDK 64-Bit Server VM (build 24.95-b01, mixed mode)

**Comment by pdemilly** [comment]
Possibly I am getting that error because I am running it on my backend server and it is trying to open a browser instance which I don't need. I have tabula running in an iframe in our app. Maybe we could have an option to disable running a browser instance -Dstart_browser=false

**Comment by jeremybmerrill** [comment]
Interesting... That sounds like a plausible cause for the issue, but the startup script should be checking `java.awt.Desktop.isDesktopSupported` to see if your system has a desktop in which a browser can be opened.

For categorization and linking purposes, this is related to #147, #171 and https://github.com/tabulapdf/tabula/commit/ddd6b31549902433aeb6456db7d592d6e6331dfc

As a quick-and-dirty workaround, you can rename the jar to anything other than "tabula.jar" and it won't try to open the browser. The CLI argument idea is a good one as a better solution that magic filenames though :)

@mtigas: If you can remember what went into fixing this 2 years ago, any idea on why it's cropping up again?

**Comment by mtigas** [comment]
Took a quick look at this and alternate port numbers seem to work fine here.

I think _this_ particular bug might be due to the remote server that pretends it has a desktop but does not have a browser. There might be some bug at server init time when that fails. Or something. Will pry into it some more.

**Comment by jazzido** [comment]
This should be fixed in 1.1.0, right @mtigas ? Feel free to close if so.

**Comment by mtigas** [comment]
Ah yeah; starting in c77f53247e4715dd97a61e1813d7188d89f9d26f we skip the _entire_ check for `java.awt.Desktop.` stuff and do not try to open browser unless you set `-Dtabula.openBrowser=true`, so this issue (browser trying to open URI if you set another port or some other cause) should be fixed.
